jerry-core/parser/js/js-parser-expr.c

```diff
@@ -1504,19 +1500,14 @@ parser_process_unary_expression (parser_context_t *context_p) /**< context */
 
         if (is_eval)
         {
+          parser_emit_cbc (context_p, CBC_EVAL);
```

Comment:
Isn't this causing extra work for the `#ifndef CONFIG_DISABLE_ES2015_CLASS` && `context_p->status_flags & PARSER_CLASS_HAS_SUPER` case? A `CBC_EVAL` is emitted just to be overwritten (below) with `context_p->last_cbc_opcode =`.

Reply:
This is how CBC works in general. You start with an opcode and mutate it to others. While this could be done a bit more efficiently, I think the perf gain is negligible, since eval is not that frequent. And looks much better.

Reply:
Yes, bytecode rewrite happens, when a different step realizes that it can do something more efficiently than the previous step. But this is different as CBC rewrite happens right after the CBC is written. I don't think that we should write intentionally inefficient code only to avoid an `#ifndef`.

Reply:
I you mean by "manually" that it writes `last_cbc_opcode` and `last_cbc.value` directly, then that's still happening. To avoid that, the proper approach would be to add a new `parser_emit_cbc_class_eval` (or something like that) to js-parser-util.c and call that from the `if (context_p->status_flags & PARSER_CLASS_HAS_SUPER)` branch (and keep calling `parser_emit_cbc (context_p, CBC_EVAL);` otherwise).

Reply:
Code rewritten. I don't like these "Outdated" labels.